src/xhr.corexhr.js

```diff
@@ -237,11 +237,7 @@ XhrShim.prototype.send = function(data) {
   } else if (data instanceof ArrayBuffer) {
     this.xhr_.send({buffer: data});
   } else if (data instanceof Blob) {
-    var reader = new FileReader();
-    reader.onload = function() {
-      this.xhr_.send({buffer: reader.result});
-    }.bind(this);
-    reader.readAsArrayBuffer(data);
+    this.xhr_.send({blob: data});
```

Comment:
AFAIK, Blob is fine, so I don't expect to undo this change.  The integration tests prove that this works on Chrome too, and Freedom uses blobs in other core APIs (e.g. core.websocket) already.